docs/src/migration.md

```diff
@@ -0,0 +1,112 @@
+
+### Factor aliases
+
+In version `4.0`, we can define factor aliases to define different implementations of the same factor. For example, in `ReactiveMP.jl`, there are multiple implentations of the `Normal` distribution. Previously, we would have to explicitly call `NormalMeanVariance` or `NormalMeanPrecision`. In version `4.0`, we can define factor aliases to default to certain implementations when specific keyword arguments are used on construction. For example: `Normal(μ = 0, σ = 1)` will default to `NormalMeanVariance` and `Normal(μ = 0, τ = 1)` will default to `NormalMeanPrecision`. This allows users to quickly toggle between different implementations of the same factor, while keeping an implementation agnostic model definition.
+
+### Nested models
+
+The main difference between version `3.1` and `4.0` is the support for nested models. In version `4.0`, models can be nested within each other. This allows for more complex models to be built in a modular way. The following example demonstrates how to nest models in version `4.0`:
+
+```julia
+@model function kalman_filter_slice(y, prev_x, new_x, A, B, Q, P)
+    new_x ~ MvNormal(μ = A * prev_x, Σ = Q)
+    y ~ MvNormal(μ = B * x, Σ = P)
+end
+
+@model function state_space_model(y, A, B, Q, P)
+    x[1] ~ MvNormal(zeros(2), diagm(ones(2)))
+    for i in eachindex(y)
+        y[i] ~ kalman_filter_slice(prev_x = x[i], new_x = new(x[i + 1]), A=A, B=B, Q=Q, P=P)
+    end
+end
+```
+Note that we reuse the `kalman_filter_slice` model in the `state_space_model` model. In the argument list of any `GraphPPL` model, we have to specify the Markov Blanket of the model we are defining. This means that all interfaces with the outside world have to be passed as arguments to the model. For the `kalman_filter_slice` model, we pass the previous state `prev_x`, the new state `new_x`, the observation `y` as well as the parameters `A`, `B`, `Q` and `P`. This means that, when invoking a submodel in a larger model, we can specify all components of the Markov Blanket. Note that, in the `state_space_model` model, we do not pass `y` as an argument to the `kalman_filter_slice` model. `GraphPPL` will infer that `y` is missing from the argument list and assign it to whatever is left of the `~` operator. Note that we also use the `new(x[i + 1])` syntax to create a new variable in the position of `x[i + 1]`. Since `y` is also passed in the argument list of the `state_space_model` model, we could have written this line with the equivalen statement `x[i + 1] ~ kalman_filter_slice(prev_x = x[i], y = y[i], A=A, B=B, Q=Q, P=P)`. However, to respect the generative direction of the model and to make the code more readable, we use the `new(x[i + 1])` syntax.
```

Comment:
```suggestion
Note that we reuse the `kalman_filter_slice` model in the `state_space_model` model. In the argument list of any `GraphPPL` model, we have to specify the Markov Blanket of the model we are defining. This means that all interfaces with the outside world have to be passed as arguments to the model. For the `kalman_filter_slice` model, we pass the previous state `prev_x`, the new state `new_x`, the observation `y` as well as the parameters `A`, `B`, `Q` and `P`. This means that, when invoking a submodel in a larger model, we can specify all components of the Markov Blanket. Note that, in the `state_space_model` model, we do not pass `y` as an argument to the `kalman_filter_slice` model. `GraphPPL` will infer that `y` is missing from the argument list and assign it to whatever is left of the `~` operator. Note that we also use the `new(x[i + 1])` syntax to create a new variable in the position of `x[i + 1]`. Since `y` is also passed in the argument list of the `state_space_model` model, we could have written this line with the equivalen statement `x[i + 1] ~ kalman_filter_slice(prev_x = x[i], y = y[i], A=A, B=B, Q=Q, P=P)`. However, to respect the generative direction of the model and to make the code more readable, we use the `new(x[i + 1])` syntax. Note, however, that the underlaying representation of the models in `GraphPPL` are still undirected.
```